File: sdk-android/src/main/java/org/sinou/android/pydia/auth/PasswordStore.java
package org.sinou.android.pydia.auth;

import com.pydio.cells.api.Store;

import java.util.Map;

public class PasswordStore implements Store<String> {

    @Override
    public void put(String id, String password) {
        Database.addPassword(id, password);
    }

    @Override
    public String get(String id) {
        return Database.password(id);
    }

    @Override
    public void remove(String id) {
        Database.deletePassword(id);
    }

    @Override
    public void clear() {
        // We do not want to enable emptying the password DB oder ?
        throw new RuntimeException("Forbidden call");
    }

    @Override
    public Map<String, String> getAll() {
        // We do not want to enable token listing
        //         return Database.listAllTokens();
        throw new RuntimeException("Forbidden call");
    }

}

File: client-core/src/main/kotlin/org/sinou/pydia/client/core/util/PasswordStore.kt
package org.sinou.pydia.client.core.util

import org.sinou.pydia.sdk.api.Store

class MemoryPasswordStore() : Store<String> {

    private val passwords = mutableMapOf<String, String>()

    override fun put(id: String, obj: String) {
        passwords.put(id, obj)
    }

    override fun get(id: String): String? {
        return passwords.get(id)
    }

    override fun remove(id: String) {
        passwords.remove(id)
    }

    override fun clear() {
        passwords.clear()
    }

    override fun getAll(): Map<String, String> {
        return passwords
    }
}
